larsim/SimFilters/FilterSimPhotonLiteTime_module.cc

```diff
@@ -0,0 +1,170 @@
+public:
+  explicit FilterSimPhotonLiteTime(fhicl::ParameterSet const& p,
+                               art::ProcessingFrame const&);
+
+  // Plugins should not be copied or assigned.
+  FilterSimPhotonLiteTime(FilterSimPhotonLiteTime const&) = delete;
+  FilterSimPhotonLiteTime(FilterSimPhotonLiteTime&&) = delete;
+  FilterSimPhotonLiteTime& operator=(FilterSimPhotonLiteTime const&) = delete;
+  FilterSimPhotonLiteTime& operator=(FilterSimPhotonLiteTime&&) = delete;
+
+private:
+  bool filter(art::Event& e, art::ProcessingFrame const&) override;
+
+  std::string const fSimPhotonsLiteCollectionLabel; //!< Label for the sim::SimPhotonsLite data product
+  std::vector<std::vector<int>> const fTimeWindows; //!< Time windows used for filtering. Units are the same as in the sim::SimPhotonsLite
+  int const fMinTotalPhotons; //!< Minimum number of photons inside a window to pass the filter
+  bool const fDebug; //!< Set to true to print (a lot of) debug information.
+  std::size_t const fN; //!< Number of time winows.
+  bool fUseReflectedPhotons; //!< Whether to include reflected photons in the filter.
+  std::string fReflectedLabel; //!< Label for the reflected photons -- "Reflected" by default.
+
+  void CheckTimeWindows() const;
+};
+
+simfilter::FilterSimPhotonLiteTime::FilterSimPhotonLiteTime(
+  fhicl::ParameterSet const& p,
+  art::ProcessingFrame const&)
+  : SharedFilter{p}
+  , fSimPhotonsLiteCollectionLabel(p.get<std::string>("SimPhotonsLiteCollectionLabel"))
+  , fTimeWindows(p.get<std::vector<std::vector<int>>>("TimeWindows"))
+  , fMinTotalPhotons(p.get<int>("MinTotalPhotons"))
+  , fDebug(p.get<bool>("Debug", false))
+  , fN(fTimeWindows.size())
+  , fUseReflectedPhotons(p.get<bool>("UseReflectedPhotons", false))
+  , fReflectedLabel(p.get<std::string>("fReflectedLabel", "Reflected"))
+{
+  CheckTimeWindows();
+
+  // For printing out debug messages, we want to serialize the
+  // event-level calls so that the messages are not garbled.
+  // Otherwise, this module works well for asynchronous event-level
+  // calls.
+  if (fDebug) {
+    serialize();
+  } else {
+    async<art::InEvent>();
+  }
+}
+
+void
+simfilter::FilterSimPhotonLiteTime::CheckTimeWindows() const
+{
+
+  if (fDebug)
+    std::cout << "\tFilterSimPhotonLiteTime: TimeWindows size is "
+              << fTimeWindows.size() << std::endl;
+
+  for (auto const& tw : fTimeWindows) {
+    if (tw.size() != 2)
+      throw cet::exception("FilterSimPhotonLiteTime::CheckTimeWindows")
+        << "Bad time window initialization: time window has wrong size (not 2)."
+        << std::endl;
+
+    if (fDebug)
+      std::cout << "\t\tTimeWindow "
+                << "[" << tw[0] << "," << tw[1] << "]" << std::endl;
+
+    if (tw[0] > tw[1])
+      throw cet::exception("FilterSimPhotonLiteTime::CheckTimeWindows")
+        << "Bad time window initialization: tw[0]>tw[1]. Reverse the order!"
+        << std::endl;
```

Comment:
Seeing as you're being extra careful with the inputs, perhaps you should defend against the two time windows being the same?

Reply:
I don't think it's best to check this -- I agree such a configuration would be weird, but it wouldn't be strictly incorrect or cause wrong behavior.